Sources/OneWay/Extensions/AsyncSequence+Extension.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  OneWay
+//  The MIT License (MIT)
+//
+//  Copyright (c) 2022-2023 SeungYeop Yeom ( https://github.com/DevYeom ).
+//
+
+import Foundation
+
+extension AsyncSequence {
+    public func gather(_ take: Int? = nil) async throws -> [Element] {
+        var results: [Element] = []
+        for try await element in self {
+            results.append(element)
+            
+            if let take = take, results.count >= take {
+               break
+            }
+        }
+        return results
+    }
```

Comment:
It seems like a very useful function. 👍 
However, I'm not sure if what the library provides is appropriate.
I think implementing the convenience in a project that uses this library might be more flexible.

Reply:
Right! i forgot to delete the file. I'll do it
Since my first idea was using gathering first and then Merge Effect, that why i used it.